source/Depreciation.cpp

```diff
@@ -30,10 +31,10 @@ namespace {
 	// Names for the two kinds of depreciation records.
 	string NAME[2] = {"fleet depreciation", "stock depreciation"};
 	// Depreciation parameters.
-	constexpr double FULL_DEPRECIATION = 0.25;
-	constexpr double DAILY_DEPRECIATION = 0.997;
-	constexpr int GRACE_PERIOD = 7;
-	constexpr int MAX_AGE = 1000 + GRACE_PERIOD;
+	const double FULL_DEPRECIATION = GameData::GetGamerules().DepreciationMinPercent();
+	const double DAILY_DEPRECIATION = GameData::GetGamerules().DepreciationDailyPercent();
+	const int GRACE_PERIOD = GameData::GetGamerules().DepreciationGracePeriod();
+	const int MAX_AGE = GameData::GetGamerules().DepreciationMaxAge();
```

Comment:
You can't do this here. Everything in an anonymous (nameless) namespace is `static`.
This means that these values are assigned before anything is actually loaded, so they don't have good values, they actually have very bad values. You'll probably need to retrieve these values on demand, with whatever performance penalties that might entail.
Another option might be to leave them in the anonymous namespace but make them non-const, and assign the values with a static method in this class that is called after the Gamerules are loaded, that can then be recalled if the gamerules are changed at runtime to update the values here.

Reply:
I have no idea where the loading are even called so I'll probably have to go with the first option.